specification/protocol/generate_rest.yaml

```diff
@@ -0,0 +1,133 @@
+openapi: 3.1.0
+info:
+  title: Open Inference API for text generation
+  description: Open Inference API for text generation
+  version: 1.0.0
+components:
+  schemas:
+    GenerateRequest:
+      type: object
+      required: 
+        - text_input
+      properties:
+        text_input:
+          type: string
+        parameters:
+          $ref: '#/components/schemas/GenerateParameters'
+    GenerateParameters:
+      type: object
+      properties:
+        parameter:
```

Comment:
This only allows one single parameter, we can probably use allOf
```yaml
parameters:
  allOf:
    - $ref: '#/components/schemas/GenerateParameters'
    - type: object
       additionalProperties:
         type: string
```

Reply:
<img width="882" alt="Screenshot 2023-12-20 at 5 15 58 PM" src="https://github.com/kserve/open-inference-protocol/assets/121221367/dc737323-1b4c-413b-9888-b94e4ef72bfe">

Just want to make sure I got it correct. Does this match the expectation?

Reply:
actually we can make the additional parameters free formed with `additionalProperties: {}`, not limited to string